cupy/core/_routines_indexing.pyx

```diff
@@ -412,10 +428,63 @@ cdef ndarray _simple_getitem(ndarray a, list slice_list):
     return v
 
 
-_nonzero_kernel_1d = ElementwiseKernel(
-    'T src, S index', 'raw U dst',
-    'if (src != 0) dst[index - 1] = i',
-    'nonzero_kernel_1d')
+_preamble = '' if not runtime._is_hip_environment else r'''
+    // ignore mask
+    #define __shfl_up_sync(m, x, y, z) __shfl_up(x, y, z)
```

Comment:
This is not for this PR, but I wonder if we should add these kinds of definitions in a generic header applied to all kernels? @leofang, what do you think?

Reply:
Yeah I can do a house keeping PR to collect them in one place. carray.cuh could be a good place to go, although it's a bit cluttered.